pkg/reconciler/events/cache/cacheclient.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"context"
+
+	lru "github.com/hashicorp/golang-lru"
+	"k8s.io/client-go/rest"
+	"knative.dev/pkg/injection"
+	"knative.dev/pkg/logging"
+)
+
+// With 4 events per Run, we can store events for 1024 concurrent Runs
+const bufferSize = 4096
+
+func init() {
+	injection.Default.RegisterClient(withCacheClient)
+}
+
+// cacheKey is a way to associate the Cache from inside the context.Context
+type cacheKey struct{}
+
+func withCacheClient(ctx context.Context, cfg *rest.Config) context.Context {
```

Comment:
there's some duplication here with the`withFakeCacheClient` func below that could probably be pulled into a helper that accepts a buffer size.

Reply:
done